cms/djangoapps/contentstore/views/tests/test_transcript_settings.py

```diff
@@ -385,6 +399,27 @@ def test_transcript_upload_handler_missing_attrs(self, request_payload, expected
         self.assertEqual(response.status_code, 400)
         self.assertEqual(json.loads(response.content)['error'], expected_error_message)
 
+    @patch('contentstore.views.transcript_settings.get_available_transcript_languages', Mock(return_value=['en', 'es']))
+    def test_transcript_upload_handler_existing_transcript(self):
+        """
+        Tests that upload handler do not update transcript's language if a transcript
+        with the same language already there for an edx_video_id.
+        """
+        transcript_upload_url = self.get_url_for_course_key(self.course.id)
+        # Make request to transcript upload handler
+        request_payload = {
+            'edx_video_id': '1234',
+            'language_code': 'en',
+            'new_language_code': 'es'
+        }
+        response = self.client.post(transcript_upload_url, request_payload, format='multipart')
+        self.assertEqual(response.status_code, 400)
+        self.assertEqual(
+            json.loads(response.content)['error'],
+            u'Transcript with "es" language code is already present.'
+        )
+
+    @patch('contentstore.views.transcript_settings.get_available_transcript_languages', Mock(return_value=['en']))
```

Comment:
Why are we mocking `en`?

Reply:
Its mocking `get_available_transcript_languages ` call to edx-VAL.